pandas/core/tools/datetimes.py

```diff
@@ -139,13 +139,16 @@ def _guess_datetime_format_for_array(arr, dayfirst: bool | None = False) -> str
             )
             if guessed_format is not None:
                 return guessed_format
-            warnings.warn(
-                "Could not infer format, so each element will be parsed "
-                "individually, falling back to `dateutil`. To ensure parsing is "
-                "consistent and as-expected, please specify a format.",
-                UserWarning,
-                stacklevel=find_stack_level(),
-            )
+            # If there are multiple non-null elements, warn about
+            # how parsing might not be consistent
+            if tslib.first_non_null(arr[first_non_null + 1 :]) != -1:
```

Comment:
Only minor nit I'd have is to add an explicit `if first_non_null == len(arr)` check preceding this. It doesn't have any impact as long as this stays a purely Python function, but if someone tried to convert down to C I'm not sure how slicing beyond the length of the array would work

Reply:
Ignore this. Thought this was a pyx file